spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java

```diff
@@ -135,6 +137,8 @@
 @DirtiesContext
 public class EnableIntegrationTests {
 
+	private static ThreadLocal<Boolean> lifecycleStopped = new ThreadLocal<>();
```

Comment:
I don't like this `ThreadLocal` approach, so I've change it to this https://github.com/artembilan/spring-integration/commit/55aa98eb93d3601af74ede0aa63f6fd9e1095cb2
Let me know and I'll merge